### ISSUE TITLE
usable

### ISSUE THREAD

**Issue by joeblew99** [mkustermann/grpc-dart-poc#1]
Hey

i am a golang programmer, but for the last 2 months been working on flutter and angular 2 in dart.

i am seriously needing a way for Dart to interoperate with golang.
GRPC really seems like the way forward i feel.

But this is the only library i have found.
Is this stable / usable yet ?

**Comment by mkustermann** [comment]
Hi Joe,

as I described in the readme file, this is a proof-of-concept. One thing missing is grpc compression. The http2 implementation it uses is missing stream prioritization.

That being said, one goal of the grpc team was that it should be **very easy** adapt to new languages (given an http2 implementation). So it might not take much more code than this proof of concept.

**Comment by joeblew99** [comment]
sounds like i can try it out.

I doubt it wil work in a browser, because of the lack of header / framing support in the browsers. 
Normally you put a gateway in front. See grpc gateway - https://github.com/grpc-ecosystem/grpc-gateway.
Did you try using it in a browser ?

But it looks like it will work server to server

**Comment by mkustermann** [comment]
I agree, it won't work inside a browser due to the lack of normal socket support. It's the first time I hear of _grpc-gateway_, so I don't know how it works. But I doubt that it will be fully compatible with normal grpc due to the lack of streaming.

**Comment by joeblew99** [comment]
thanks for the answer.

I am thinking now of a better way to get dart to talk to golang that is low friction.
stdio is pretty horrible.

Here are my scenarios:
This scenarios i know how to do:
I have a Dart client (web thing wrapped in electron).
I have a golang server, that is just publishing web sockets and restful json

This scenario i dont knwo how to do:
I have a Dart client (web thing wrapped in electron).
I have a dart server, that is publishing web sockets and restful json. I woudl prefer a nice RPC JSON though :)
I have a golang application server, that is called by the dart server. No idea how to allow them to do RPC.

If you have any ideas please let me know. I will close this issue if i am bugging you too much of course...

**Comment by ghost** [comment]
here is the link for you:
https://github.com/grpc-ecosystem/

The gateway is in there

websocket support:
https://github.com/tmc/grpc-websocket-proxy

**Comment by joeblew99** [comment]
@mkustermann There is now an agreed spec for GRPC-web

So a Dart implementation is now doable that will work on FLutter and Dart web.


https://github.com/grpc/grpc/issues/8682
See the top link for the link to the Spec.

I would like to help on Te Dart code.
